README.rst

```diff
@@ -27,7 +27,7 @@ planar geometric objects. It is based on the widely deployed `GEOS
 libraries. Shapely is not concerned with data formats or coordinate systems,
 but can be readily integrated with packages that are. For more details, see:
 
-* `Shapely GitHub repository <https://github.com/Toblerity/Shapely>`__
+* `Shapely GitHub repository <https://github.com/shapely/shapely>`__
 * `Shapely documentation and manual <https://shapely.readthedocs.io/en/latest/>`__
```

Comment:
There's one outstanding task in #1204 which would enable this link to be updated to the correct version of documentation.

Reply:
https://shapely.readthedocs.io/en/stable/ url seems to work?

Reply:
I was looking for https://shapely.readthedocs.io/en/maint-1.8/ and just realised that it works! But wasn't a link to click in RTD for some reason.

Reply:
It works because I just activated both maint-1.8.0 branch and 1.8.0 tag ;) 
The link should appear in the RTD versions box, but you might need a browser hard page refresh to get those (I suppose the available versions are cached)